### PR TITLE
feat(forms): add tooltip to widgets

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

It's impossible to show tooltips on form fields

**What is the chosen solution to this problem?**

Add a new `tooltip` prop and display it with a tooltip trigger 

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -8,5 +8,8 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fieldsets/Array/ArrayItem.component.js
   3:8  error  'Icon' is defined but never used  no-unused-vars
 
-✖ 2 problems (2 errors, 0 warnings)
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/Widget/Widget.component.js
+  9:32  error  Absolute imports should come before relative imports  import/imports-first
```
Comment (frassinier) [inline output/forms.eslint.txt]: ⚠️

File: packages/forms/src/UIForm/Widget/Widget.component.js [hunk 2]
```diff
@@ -47,6 +57,25 @@ export default function Widget(props) {
 	const error = getError(props.errors, props.schema);
 	const errorMessage = validationMessage || error;
 
+	if (tooltip) {
+		return (
+			<TooltipTrigger label={tooltip} tooltipPlacement={'left'}>
```
Comment (frassinier) [inline packages/forms/src/UIForm/Widget/Widget.component.js]: I'm not convinced by the position, it fits in TDP but we should let the possibility to change it

File: packages/forms/src/UIForm/Widget/Widget.component.js [hunk 3]
```diff
@@ -47,6 +57,25 @@ export default function Widget(props) {
...
+				<div>
+					<WidgetImpl
```
Comment (frassinier) [inline packages/forms/src/UIForm/Widget/Widget.component.js]: Please DRY, JSX should be in a function `{getWidgetImpl()}`

File: packages/forms/src/UIForm/Widget/Widget.component.js [hunk 4]
```diff
@@ -47,6 +57,25 @@ export default function Widget(props) {
 	const error = getError(props.errors, props.schema);
 	const errorMessage = validationMessage || error;
 
+	if (tooltip) {
+		return (
+			<TooltipTrigger label={tooltip} tooltipPlacement={'left'}>
+				<div>
```
Comment (frassinier) [inline packages/forms/src/UIForm/Widget/Widget.component.js]: May we can explain why we've chosen to use div here in a comment?

File: packages/forms/src/UIForm/Widget/Widget.component.test.js [hunk 5]
```diff
@@ -41,6 +41,22 @@ describe('Widget component', () => {
...
+				onChange={jest.fn('onChange')}
+				onFinish={jest.fn('onFinish')}
+				onTrigger={jest.fn('onTrigger')}
+				properties={properties}
+				schema={{ ...schema, tooltip: 'example tooltip' }}
+				errors={errors}
+			/>,
+		);
+
+		expect(wrapper.getElement()).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/Widget/Widget.component.test.js]: please replace toMatchSnapshot with check wrapper.getElement().type is TooltipTrigger with the good label (props() on it)
Nobody read snapshots

File: packages/forms/src/UIForm/Widget/Widget.component.js (hunk 2 above)
Comment (jmfrancois) [reply packages/forms/src/UIForm/Widget/Widget.component.js]: +1

File: packages/forms/src/UIForm/Widget/Widget.component.js (hunk 4 above)
Comment (frassinier) [reply packages/forms/src/UIForm/Widget/Widget.component.js]: ℹ️We need an extra `<div>` here to get a wrapper with a `display: block`

File: packages/forms/src/UIForm/Widget/Widget.component.js [hunk 6]
```diff
@@ -89,6 +109,7 @@ if (process.env.NODE_ENV !== 'production') {
 
 Widget.defaultProps = {
 	widgets: {},
+	tooltipPlacement: 'left',
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/Widget/Widget.component.js]: please remove this default and use || 'left'

**Pull request merged**